### PR TITLE
Added tracking scratch progress

### PR DESCRIPTION
**Pull request by Ra1nWarden**
I am trying to add tracking of scratching progress delegate using a boolean array. (#2)
Example is updated with delegate method.

### CODE REVIEW

File: STScratchView/STScratchView.h [hunk 1]
```diff
@@ -6,10 +6,20 @@
 //  Copyright (c) 2012 Sebastien Thiebaud. All rights reserved.
 //
 
+@class STScratchView;
+
+@protocol STScratchViewDelegate
+@required
+
+- (void)STScratchView:(STScratchView*)scratchView didChangeProgress:(CGFloat)prog;
```
Comment (SebastienThiebaud) [inline STScratchView/STScratchView.h]: Why is it required? rename `prog` to `progress`

File: STScratchView/STScratchView.h [hunk 2]
```diff
@@ -6,10 +6,20 @@
...
+
+@end
+
 @interface STScratchView : UIView
 
 @property (nonatomic, assign) float sizeBrush;
 @property (nonatomic, strong) UIView *hideView;
+@property (nonatomic, assign) id<STScratchViewDelegate> delegate;
```
Comment (SebastienThiebaud) [inline STScratchView/STScratchView.h]: Must be `weak`, not `assign`

File: STScratchView/STScratchView.m [hunk 3]
```diff
@@ -86,6 +93,16 @@ - (void)setHideView:(UIView *)hideView
     
     CGImageRelease(mask);
     CGColorSpaceRelease(colorspace);
+
+    _width = imageWidth / _sizeBrush;
```
Comment (SebastienThiebaud) [inline STScratchView/STScratchView.m]: Where did you get that code, any link please?

File: STScratchView/STScratchView.m [hunk 4]
```diff
@@ -95,6 +112,31 @@ - (void)scratchTheViewFrom:(CGPoint)startPoint to:(CGPoint)endPoint
     CGContextMoveToPoint(_contextMask, startPoint.x * scale, (self.frame.size.height - startPoint.y) * scale);
 	CGContextAddLineToPoint(_contextMask, endPoint.x * scale, (self.frame.size.height - endPoint.y) * scale);
 	CGContextStrokePath(_contextMask);
+
```
Comment (SebastienThiebaud) [inline STScratchView/STScratchView.m]: Same here.

**Comment by SebastienThiebaud** [comment]
See my code notes and also can you squash your commits, it's a bit nasty. Thanks.

**Comment by Ra1nWarden** [comment]
Thanks for the review. I was inspired by this [project](https://github.com/moqod/ios-scratch-n-see). I was trying to mimic how it was done there.

**Comment by Ra1nWarden** [comment]
See squashed PR #9

**Pull request closed without merging**